### PR TITLE
Consts cleanup

### PR DESCRIPTION
**Pull request by tal-nc**
cleaned up consts. returned caching and overridden hash and eq.
removed un-used sub_matrix.py

### CODE REVIEW

File: src/corona_hakab_model/consts.py [hunk 1]
```diff
@@ -116,18 +117,7 @@
...
-    defaults=[default_parameters[key] for key in sorted(default_parameters)],
-)
-
-
-class Consts(ConstParameters):
-    __slots__ = ()
+    }
 
     @staticmethod
     def from_file(param_path):
```
Comment (bentheiii) [inline src/corona_hakab_model/consts.py:123]: should be a classmethod that calls cls

File: src/corona_hakab_model/consts.py [hunk 2]
```diff
@@ -140,19 +130,21 @@ def from_file(param_path):
...
+        parameters = eval(
+            data,
+            {
+                "__builtins__": None,
+                "dist": dist,
+                "rv_discrete": rv_discrete,
+                "DetectionTest": DetectionTest,
+                "ConditionedPolicy": ConditionedPolicy,
+                "ConnectionTypes": ConnectionTypes,
+            },
```
Comment (bentheiii) [inline src/corona_hakab_model/consts.py:142]: maybe this dictionary is too big to put in a call? place it in a separate variable?

File: src/corona_hakab_model/consts.py [hunk 3]
```diff
@@ -212,6 +204,7 @@ def hospitalized_to_icu_probability(self) -> float:
     def icu_to_dead_probability(self) -> float:
         return 1 - self.icu_to_hospitalized_probability
 
+    @lru_cache
```
Comment (bentheiii) [inline src/corona_hakab_model/consts.py:207]: should be lru_cache(None)

File: src/corona_hakab_model/consts.py [hunk 4]
```diff
@@ -277,28 +270,15 @@ class ImmuneTerminalState(ImmuneState, TerminalState):
...
-
-    @property
-    # todo this should be a consts
-    def non_circular_matrices(self):
-        return [
-            NonCircularConnectionMatrix("strangers", None, self.strangers_scale_factor, self.stranger_strength),
-        ]
+    # overriding hash and eq to allow caching while using un-hashable attributes
+    def __hash__(self):
+        return object.__hash__(self)
```
Comment (bentheiii) [inline src/corona_hakab_model/consts.py:275]: do __hash__ = object.__hash__ instead (faster)

File: src/corona_hakab_model/consts.py [hunk 5]
```diff
@@ -277,28 +270,15 @@ class ImmuneTerminalState(ImmuneState, TerminalState):
...
 
-    @property
-    def clustered_matrices(self):
-        return [
-            ClusteredConnectionsMatrix("work", None, self.work_scale_factor, self.work_strength),
-            ClusteredConnectionsMatrix("school", None, self.school_scale_factor, self.school_strength),
-        ]
+    def __eq__(self, other):
```
Comment (bentheiii) [inline src/corona_hakab_model/consts.py:277]: ditto

**Comment by bentheiii** [comment]
did you check that this runs?

File: src/corona_hakab_model/consts.py (hunk 1 above)
Comment (tal-nc) [reply src/corona_hakab_model/consts.py:123]: done

File: src/corona_hakab_model/consts.py (hunk 2 above)
Comment (tal-nc) [reply src/corona_hakab_model/consts.py:142]: kk

File: src/corona_hakab_model/consts.py (hunk 3 above)
Comment (tal-nc) [reply src/corona_hakab_model/consts.py:207]: done

File: src/corona_hakab_model/consts.py (hunk 4 above)
Comment (tal-nc) [reply src/corona_hakab_model/consts.py]: done

File: src/corona_hakab_model/consts.py (hunk 5 above)
Comment (tal-nc) [reply src/corona_hakab_model/consts.py]: done

**Comment by tal-nc** [comment]
i checked it, and it works.
checked calling consts with key words and it also works.
only thing i didn't check is importing consts from a file.

**Pull request merged**